### ISSUE TITLE
GHAS Compliance Action Default Branch Limitation

### ISSUE THREAD

**Issue by HaleenUptain** [GeekMasher/advanced-security-compliance#72]
### Description
Limitation identified using Code Scanning and Secret Scanning checks, they are only supported on the default branch.  We had hoped these checks could be performed on branch pushes and PRs to catch alerts before they are propagated to the default branch but that functionality does not seem to be supported.

### Propose Solution
Working with this action we like what we see.  We are very interested in this concept and the ability to push security checks farther left in the development process.  Proposed solution is to modify this action to work on any branch, not just the default branch, so checks catch alerts on branch pushes and PRs before they are propagated to the default branch.

**Comment by GeekMasher** [comment]
@HaleenUptain You are right and this is a missing functionality. This was intended but is missing.

https://github.com/GeekMasher/advanced-security-compliance/blob/main/ghascompliance/octokit/octokit.py#L54

How important is this functionality to you? I can spend some time this / next week building this out as it would take long to build out

**Comment by HaleenUptain** [comment]
Ability to run a GHAS Compliance action against non-default branches to stop alerts before they are merged into the default branch is critical functionality we are looking for in a GHAS Compliance action.  If we could at least get that working for "Code Scanning" that would help and be a big step forward.

Looks like the passed in ref is never used.
- https://github.com/GeekMasher/advanced-security-compliance/blob/main/ghascompliance/octokit/codescanning.py#L8

Our tests also indicate that ref is not used.  We ran a couple bare minimum tests to prove this out. Simple stand alone compliance job using workflow_dispatch to run the jobs manually.  Note:  Code scanning is the only compliance feature we have turned on in these tests.

**Default Branch Test - main**
main branch has no high alerts.  Policy snippet:
codescanning:
  level: error
Standalone compliance job running in main reports that correctly.  Ref in .yml is set to:
ref: refs/heads/main

![image](https://user-images.githubusercontent.com/97243098/206003235-999b7e7f-e564-441c-82c1-06c4e2fdf55f.png)

![image](https://user-images.githubusercontent.com/97243098/206003903-ed717375-0ea6-4ffe-92c1-5a4ec0f0bfed.png)

**Non-Default Branch Test - featurebranch_test**
featurebranch_test has one high alert.
Compliance job running in main is not returning expected result when we change ref in .yml to point to a non-default branch.  We would expect the high alert on featurebranch_test branch to show as a threshold violation:
ref: refs/heads/featurebranch_test
![image](https://user-images.githubusercontent.com/97243098/206005421-6597d85f-be69-4e0c-9706-da81d9e811db.png)

![image](https://user-images.githubusercontent.com/97243098/206005789-51150445-9435-4eed-ac4d-4af772d67596.png)

**Comment by GeekMasher** [comment]
@HaleenUptain I have moved this repository to a more official repository https://github.com/advanced-security/policy-as-code .

This issue is top priority for myself so I will be working on in before the holidays, please check the new issue opened here

https://github.com/advanced-security/policy-as-code/issues/2
